app/controllers/chargeback_controller.rb

```diff
@@ -273,9 +300,136 @@ def cb_rate_form_field_changed
     cb_rate_get_form_vars
   end
 
+  # AJAX driven routine to storage the selected rate details for remove
+  def cb_selected_rd
+    @edit = session[:edit]
+    unless params[:select_all].nil?
+      if params[:select_all] != "null"
+        @edit[:selected_rd_remove] = [*0..(@sb[:rate_details].select { |k| k.rate != "0" }).size - 1]
+        enable_button = true
+      else
+        @edit[:selected_rd_remove] = []
+        enable_button = false
+      end
+      render :update do |page|
+        if enable_button
+          page << javascript_enable_field("btm-remove_rd")
+          page << javascript_checked_by_class("select")
+        else
+          page << javascript_disable_field("btm-remove_rd")
+          page << javascript_unchecked_by_class("select")
+        end
+      end
+    end
+
+    unless params[:select].nil?
+      if params[:select] != "null"
+        enable_button = true
+        if @edit[:selected_rd_remove].nil?
+          @edit[:selected_rd_remove] = [params[:select].to_i]
+        else
+          @edit[:selected_rd_remove] = @edit[:selected_rd_remove].append(params[:select].to_i)
+        end
+      else
+        @edit[:selected_rd_remove].delete(params[:index].to_i)
+        @edit[:selected_rd_remove].nil? ? enable_button = false : enable_button = true
+      end
+      render :update do |page|
+        if enable_button
+          page << javascript_enable_field("btm-remove_rd")
+        else
+          page << javascript_disable_field("btm-remove_rd")
+        end
+        page << javascript_unchecked("select_all")
+        page << javascript_checked_by_class("select[value=#{params[:index].to_i}]")
+      end
+    end
+  end
+
+  # AJAX driven routine to check for changes in ANY field on the form off add rate detail in rate
+  def cb_add_rd_form_field_changed
+    @edit = session[:edit]
+    render :update do |page|
+      if !params[:metric].nil?
```

Comment:
Same here, can multiple params be changed at once?  Either way, see if extracting each condition to a method helps to refactor any common logic each method ends up doing.